### PR TITLE
removed some `RowsOfMatrix(ImmutableMatrix(...))` calls

### PR DESCRIPTION
**Pull request by ThomasBreuer**
This addresses issue #5218.

### CODE REVIEW

File: lib/vspcrow.gi [hunk 1]
```diff
@@ -781,11 +792,11 @@ InstallOtherMethod( SemiEchelonBasisNC,
     IsIdenticalObj,
     [ IsGaussianRowSpace, IsList ],
     function( V, gens )
-    local B,  # the basis, result
-          gensi; # immutable copy
+    local B;  # the basis, result
 
     # filter for vector objects, not compressed FF vectors
     if not ForAll(gens,x->IsVectorObj(x) and not IsDataObjectRep(x)) then
+      # We expect that the method for `IsMatrix` strikes.
```
Comment (fingolfin) [inline lib/vspcrow.gi:799]: "strikes"?

File: lib/vspcrow.gi [hunk 2]
```diff
@@ -701,12 +701,15 @@ InstallOtherMethod( SemiEchelonBasis,
...
           B,       # the basis, result
+          F,       # base domain
           gensi,   # immutable copy
           flag,
           v;       # loop over vector space generators
 
     flag:=false;
     if ForAll(gens,x->IsVectorObj(x) and not IsDataObjectRep(x)) then
+      # What is meant here:
+      # 'gens' is a list of vector objects that are not nec. lists.
```
Comment (fingolfin) [inline lib/vspcrow.gi:712]:
```suggestion
      # 'gens' is a list of vector objects that are not necessarily lists.
```

File: lib/vspcrow.gi [hunk 3]
```diff
@@ -730,8 +733,17 @@ InstallOtherMethod( SemiEchelonBasis,
       SetIsRectangularTable( B, true );
     fi;
     SetUnderlyingLeftModule( B, V );
-    gensi := ImmutableMatrix(LeftActingDomain(V), gens);
-    if flag then gensi:=RowsOfMatrix(gensi);fi;
+    F:= LeftActingDomain( V );
+    if flag then
+      # In the case of proper vector objects,
+      # we want to keep their representation
+      # (since the user had good reason the give us these objects)
```
Comment (fingolfin) [inline lib/vspcrow.gi:740]: Maybe this?
```suggestion
      # (since the user had good reason to give us these objects)
```

**Review by fingolfin — APPROVED** [review]
Looks good to me though very minor tweak and question

**Comment by fingolfin** [comment]
@ThomasBreuer do you want to address any of the minor nitpicks? If not, I'll just merge this as is, but wanted to give you a chance to react :-)

**Comment by ThomasBreuer** [comment]
I will address your comments.

**Pull request merged**